test/evp_test.c

```diff
@@ -689,23 +681,24 @@ static int cipher_test_enc(EVP_TEST *t, int enc,
         t->err = "CIPHERFINAL_ERROR";
         goto err;
     }
-    if (!compare_mem(out, out_len, tmp + out_misalign, tmplen + tmpflen)) {
+    if (!TEST_mem_eq(wanted, wanted_len,
+                     tmp + out_misalign, tmplen + tmpflen)) {
```

Comment:
Suggestion to rename `*dat` to `wanted` presents challenge here. One can use `_wanted` or `expected`...